source/module_io/to_wannier90_lcao_in_pw.cpp

```diff
@@ -196,70 +211,23 @@ psi::Psi<std::complex<double>>* toWannier90_LCAO_IN_PW::get_unk_from_lcao(
+void toWannier90_LCAO_IN_PW::nao_G_expansion(
     const int& ik,
-    const ModulePW::PW_Basis_K* wfc_basis,
-    const Structure_Factor& sf,
-    ModuleBase::ComplexMatrix& psi,
-    const ModuleBase::realArray& table_local
+    const ModulePW::PW_Basis_K* wfcpw,
+    ModuleBase::ComplexMatrix& psi
 )
 {
-    ModuleBase::TITLE("toWannier90_LCAO_IN_PW", "produce_local_basis_in_pw");
-    assert(ik >= 0);
-    const int npw = wfc_basis->npwk[ik];
-    const int total_lm = (GlobalC::ucell.lmax + 1) * (GlobalC::ucell.lmax + 1);
-    ModuleBase::matrix ylm(total_lm, npw);
-
-    ModuleBase::Vector3<double>* gk = new ModuleBase::Vector3<double>[npw];
-    for (int ig = 0; ig < npw; ig++)
-    {
-        gk[ig] = wfc_basis->getgpluskcar(ik, ig);
-    }
-
-    ModuleBase::YlmReal::Ylm_Real(total_lm, npw, gk, ylm);
-
-    double* flq = new double[npw];
-    int iwall = 0;
-    for (int it = 0; it < GlobalC::ucell.ntype; it++)
+    int npwx = wfcpw->npwk_max;
+    psi::Psi<std::complex<double>>* psig = this->psi_init_->cal_psig(ik);
+    int nbands = GlobalV::NLOCAL;
+    int nbasis = npwx*GlobalV::NPOL;
+    for (int ib = 0; ib < nbands; ib++)
     {
-        for (int ia = 0; ia < GlobalC::ucell.atoms[it].na; ia++)
+        for (int ig = 0; ig < nbasis; ig++)
         {
-            std::complex<double>* sk = sf.get_sk(ik, it, ia, wfc_basis);
-            int ic = 0;
-            for (int L = 0; L < GlobalC::ucell.atoms[it].nwl + 1; L++)
-            {
-                std::complex<double> lphase = pow(ModuleBase::NEG_IMAG_UNIT, L); // mohan 2010-04-19
-                for (int N = 0; N < GlobalC::ucell.atoms[it].l_nchi[L]; N++)
-                {
-                    for (int ig = 0; ig < npw; ig++)
-                    {
-                        flq[ig] = ModuleBase::PolyInt::Polynomial_Interpolation(table_local,
-                                                                                it,
-                                                                                ic,
-                                                                                GlobalV::NQX,
-                                                                                GlobalV::DQ,
-                                                                                gk[ig].norm() * GlobalC::ucell.tpiba);
-                    }
-
-                    for (int m = 0; m < 2 * L + 1; m++)
-                    {
-                        const int lm = L * L + m;
-                        for (int ig = 0; ig < npw; ig++)
-                        {
-                            psi(iwall, ig) = lphase * sk[ig] * ylm(lm, ig) * flq[ig];
-                        }
-                        ++iwall;
-                    }
-
-                    ++ic;
-                } // end for N
-            }     // end for L
-            delete[] sk;
-        } // end for ia
-    } // end for it
-
-    delete[] flq;
-    delete[] gk;
+            psi(ib, ig) = psig[0](ik, ib, ig);
```

Comment:
I would like to ask, after converting nao to pw, when "NSPIN=4", whether the first half of the pw basis is "up", and the second half is "down" accordingly.